standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3905,110 +3908,95 @@ protected static String compactorStateToResponse(char s) {
-        }
-        //todo: sort by cq_id?
-        //what I want is order by cc_end desc, cc_start asc (but derby has a bug https://issues.apache.org/jira/browse/DERBY-6013)
-        //to sort so that currently running jobs are at the end of the list (bottom of screen)
-        //and currently running ones are in sorted by start time
-        //w/o order by likely currently running compactions will be first (LHS of Union)
-
-        String query = sb.toString();
-        dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
-        stmt = dbConn.prepareStatement(query);
-        if (org.apache.commons.lang3.StringUtils.isNotBlank(rqst.getPoolName())) {
-          stmt.setString(1, rqst.getPoolName());
-          stmt.setString(2, rqst.getPoolName());
-        }
-
-        LOG.debug("Going to execute query <{}>", query);
-        ResultSet rs = stmt.executeQuery();
-        while (rs.next()) {
-          ShowCompactResponseElement e = new ShowCompactResponseElement();
-          e.setDbname(rs.getString(1));
-          e.setTablename(rs.getString(2));
-          e.setPartitionname(rs.getString(3));
-          e.setState(compactorStateToResponse(rs.getString(4).charAt(0)));
-          try {
-            e.setType(dbCompactionType2ThriftType(rs.getString(5).charAt(0)));
-          } catch (MetaException ex) {
-            //do nothing to handle RU/D if we add another status
-          }
-          e.setWorkerid(rs.getString(6));
-          long start = rs.getLong(7);
-          if (!rs.wasNull()) {
-            e.setStart(start);
-          }
-          long endTime = rs.getLong(8);
-          if (endTime != -1) {
-            e.setEndTime(endTime);
-          }
-          e.setRunAs(rs.getString(9));
-          e.setHadoopJobId(rs.getString(10));
-          e.setId(rs.getLong(11));
-          e.setErrorMessage(rs.getString(12));
-          long enqueueTime = rs.getLong(13);
-          if (!rs.wasNull()) {
-            e.setEnqueueTime(enqueueTime);
-          }
-          e.setWorkerVersion(rs.getString(14));
-          e.setInitiatorId(rs.getString(15));
-          e.setInitiatorVersion(rs.getString(16));
-          long cleanerStart = rs.getLong(17);
-          if (!rs.wasNull() && (cleanerStart != -1)) {
-            e.setCleanerStart(cleanerStart);
-          }
-          String poolName = rs.getString(18);
-          if (org.apache.commons.lang3.StringUtils.isBlank(poolName)) {
-            e.setPoolName(DEFAULT_POOL_NAME);
-          } else {
-            e.setPoolName(poolName);
+      ShowCompactResponse response = new ShowCompactResponse(new ArrayList<>());
+      try (Connection dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED)) {
+        StringBuilder query = new StringBuilder(TxnQueries.SHOW_COMPACTION_QUERY).
+          append(getShowCompactionFilterClause(rqst)).append(TxnQueries.SHOW_COMPACTION_ORDERBY_CLAUSE);
+        try (PreparedStatement stmt = sqlGenerator.prepareStmtWithParameters(dbConn, query.toString(),
```

Comment:
as mentioned before, please move dbConn in the same try-with resources block, ie
````
try (Connection dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
PreparedStatement stmt = sqlGenerator.prepareStmtWithParameters(dbConn, query.toString(),...) {
...
}
````